sqlmesh/core/schema_diff.py

```diff
@@ -523,25 +526,33 @@ def _alter_operation(
         current_type = exp.DataType.build(current_type, copy=False)
         if self.support_nested_operations:
             if new_type.this == current_type.this == exp.DataType.Type.STRUCT:
-                return self._get_operations(
+                operations = self._get_operations(
                     columns,
                     current_type,
                     new_type,
                     root_struct,
                 )
+                if not operations[0].op == TableAlterOperationType.DROP or self.support_nested_drop:
```

Comment:
I'm a little concerned that we rely on a specific order in which operations are added to the list in `_get_operations`. Should we check the entire list instead?

Reply:
Yes, added a function to perform this check and moved it before `_get_operations` since I noticed that it mutates the current_type